core/server/services/route/settings.js

```diff
@@ -0,0 +1,65 @@
+'use-scrict';
+
+const yaml = require('js-yaml'),
+    fs = require('fs-extra'),
+    path = require('path'),
+    debug = require('ghost-ignition').debug('settings'),
+    common = require('../../lib/common'),
+    _ = require('lodash'),
+    config = require('../../config');
+
+function parseYaml(filePath, fileName) {
+    return fs.readFile(filePath, 'utf8')
+        .then((file) => {
+            try {
+                const parsed = yaml.safeLoad(file);
+                return Promise.resolve(parsed);
+            } catch (err) {
+                // Parsing failed, `js-yaml` tells us exactly what and where in the
+                // `reason` property as well as in the message.
+                return Promise.reject(new common.errors.IncorrectUsageError({
+                    message: common.i18n.t('errors.services.route.settings.error', {file: fileName, context: err.reason}),
+                    context: err.message,
+                    err: err,
+                    help: common.i18n.t('errors.services.route.settings.help', {file: fileName})
+                }));
+            }
+        }).catch((error) => {
+            return Promise.reject(error);
+        });
+}
+
+module.exports = function loadSettings() {
+    const settingsPromises = {};
+
+    // read all files from /content/settings
+    return fs.readdir(config.getContentPath('settings')).then((files) => {
+        debug('Found settings files:', files);
+
+        _.each(files, function (file) {
+            const settingsType = file.replace(/.ya?ml$/i, '');
+
+            if (file.match(/.ya?ml$/i)) {
+                settingsPromises[settingsType] = parseYaml(path.join(config.getContentPath('settings'), file), file);
```

Comment:
What kind of Object do we expect? I did it now in a way, that the filename (without extension) will be the first property and this will then have the parsed yaml file as property 🤷‍♀️